### PR TITLE
Add Support for Canary API Version in FB CAPI

### PR DESCRIPTION
**Pull request by marinhero**
# Summary

We are looking to update our FB Marketing API version from V12.0 to V14.0. This PR adds an intermediate (and safer) step towards that migration. By adding canary support we will be able to granularly control the levels of traffic received by X version of the API.

## Testing

Screen Shot 2022-07-22 at 3 09 01 PM

```
➜  quasar git:(master) treb deploy -e production -b 32696a5 integrations-qa-control-prod-2

To see the changes you will deploy, navigate to:
https://github.com/segmentio/integrations/compare/8ac2432cdcfab3cbb1b3ef477bd16667808bb2fe...32696a5

Deploy to Production? [Yn]: y
Deploying integrations-qa-control-prod-2@32696a5 to production...
Cloudformation stack 'arn:aws:cloudformation:us-west-2:752180062774:stack/production--integrations-qa-control-prod-2/b7d4fbd0-54d4-11ea-a0c6-06e1bfa6e222' updating
Updated TaskDefinition 'arn:aws:ecs:us-west-2:752180062774:task-definition/productionintegrations-qa-control-prod-2:63'
Updated Service 'arn:aws:ecs:us-west-2:752180062774:service/outbound/production-integrations-qa-control-prod-2-Service-KLN9ZN0UTXJ4'
Cloudformation stack 'arn:aws:cloudformation:us-west-2:752180062774:stack/production--integrations-qa-control-prod-2/b7d4fbd0-54d4-11ea-a0c6-06e1bfa6e222' updated

Success!
➜  quasar git:(master) treb deploy -e production -b 32696a5 integrations-qa-candidate-prod-2

To see the changes you will deploy, navigate to:
https://github.com/segmentio/integrations/compare/b73fe4e84d3a7c4d36d31dcb2f55555305f65366...32696a5

Deploy to Production? [Yn]: y
Deploying integrations-qa-candidate-prod-2@32696a5 to production...
Cloudformation stack 'arn:aws:cloudformation:us-west-2:752180062774:stack/production--integrations-qa-candidate-prod-2/740f53b0-4452-11ea-8948-02e1301110e6' updating
Updated TaskDefinition 'arn:aws:ecs:us-west-2:752180062774:task-definition/productionintegrations-qa-candidate-prod-2:92'
Updated Service 'arn:aws:ecs:us-west-2:752180062774:service/outbound/production-integrations-qa-candidate-prod-2-Service-1JTR5VATE90TW'
Cloudformation stack 'arn:aws:cloudformation:us-west-2:752180062774:stack/production--integrations-qa-candidate-prod-2/740f53b0-4452-11ea-8948-02e1301110e6' updated

Success!
➜  quasar git:(master) quasar create -e production -s 2 -i actions-facebook-conversions-api
Success! New experiment created with id: 2CJLql7Wq0ZJib2lDOvfYQBXFy9
View experiment results here: https://segment.datadoghq.com/dashboard/kpu-nnr-kn2/quasar-destinations-qa?live=true&tpl_var_candidate_service_name=integrations-qa-candidate-2&tpl_var_control_service_name=integrations-qa-control-2&tpl_var_experiment_id=2cjlql7wq0zjib2ldovfyqbxfy9&tpl_var_env=production
To view any differences, run: quasar get-diffs -e production -x 2CJLql7Wq0ZJib2lDOvfYQBXFy9
To stop experiment early, run: quasar cancel -e production -x 2CJLql7Wq0ZJib2lDOvfYQBXFy9
➜  quasar git:(master) quasar get-diffs -e production -x 2CJLql7Wq0ZJib2lDOvfYQBXFy9
Fetching diffs, this could take a minute or more for diffs over 1000
➜  quasar git:(master) quasar get-diffs -e production -x 2CJLql7Wq0ZJib2lDOvfYQBXFy9
➜  quasar git:(master) quasar create -e production -s 2 -t 120m -i actions-facebook-conversions-api
Success! New experiment created with id: 2CJVQxrfncmQvuUjTLs36msUgO6
View experiment results here: https://segment.datadoghq.com/dashboard/kpu-nnr-kn2/quasar-destinations-qa?live=true&tpl_var_candidate_service_name=integrations-qa-candidate-2&tpl_var_control_service_name=integrations-qa-control-2&tpl_var_experiment_id=2cjvqxrfncmqvuujtls36msugo6&tpl_var_env=production
To view any differences, run: quasar get-diffs -e production -x 2CJVQxrfncmQvuUjTLs36msUgO6
To stop experiment early, run: quasar cancel -e production -x 2CJVQxrfncmQvuUjTLs36msUgO6
```

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/facebook-conversions-api/__tests__/initiateCheckout.test.ts [hunk 1]
```diff
@@ -57,7 +58,7 @@ describe('FacebookConversionsApi', () => {
     })
 
     it('should throw an error for invalid currency values', async () => {
-      nock(`https://graph.facebook.com/v12.0/${settings.pixelId}`).post(`/events`).reply(201, {})
+      nock(`https://graph.facebook.com/v${API_VERSION}0/${settings.pixelId}`).post(`/events`).reply(201, {})
```
Comment (mericsson) [inline packages/destination-actions/src/destinations/facebook-conversions-api/__tests__/initiateCheckout.test.ts:61]:
```suggestion
      nock(`https://graph.facebook.com/v${API_VERSION}/${settings.pixelId}`).post(`/events`).reply(201, {})
```

?

File: packages/destination-actions/src/destinations/facebook-conversions-api/utils.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Features } from '@segment/actions-core/src/mapping-kit'
+import { API_VERSION, CANARY_API_VERSION } from './constants'
+
+export const get_api_version = (features: Features | undefined): string => {
```
Comment (mericsson) [inline packages/destination-actions/src/destinations/facebook-conversions-api/utils.ts:4]: could this / should this share a datadog metric of which version it is using?

File: packages/destination-actions/src/destinations/facebook-conversions-api/__tests__/initiateCheckout.test.ts (hunk 1 above)
Comment (rhall-twilio) [reply packages/destination-actions/src/destinations/facebook-conversions-api/__tests__/initiateCheckout.test.ts:61]: Is this test currently passing with the incorrect URL?

**Review by rhall-twilio — APPROVED** [review]

File: packages/destination-actions/src/destinations/facebook-conversions-api/utils.ts (hunk 2 above)
Comment (dlasky) [reply packages/destination-actions/src/destinations/facebook-conversions-api/utils.ts:4]: Also if you do please namespace thr metric to FB somehow so it won't conflict with future actions version reporting for fast deploys

Comment (marinhero) [reply packages/destination-actions/src/destinations/facebook-conversions-api/utils.ts]: All right, that's now part of the PR. Please take a look and let me know if it's what you had in mind.

**Comment by marinhero** [comment]
Thanks for the review @rhall-twilio! I'll wait for one more approval from @mericsson or @dlasky to make sure the latest commit meets their suggestions.

**Comment by dlasky** [comment]
could we include `fb_` as a prefix in both the tags and the stat? unless its already scoped to this destination somehow (not super great at datadog implementation tbh) I just want to make sure we wont collide when we include `version` in fast deploys stats

**Comment by marinhero** [comment]
Hey @dlasky I added a prefix to the metric. It's my understanding that tags will be scoped to the metric so I don't see the need to add the prefix there too. Please let me know if my assumptions are erroneous. Otherwise, a review would be appreciated. 

Thanks!

**Review by dlasky — APPROVED** [review]
i genuinely dont know regarding tags, but the rest of this looks good to me!

**Pull request merged**